File: generated-samples/java-v1/image-generation/key-auth/Sample.java
import com.openai.client.OpenAIClient;
import com.openai.client.okhttp.OpenAIOkHttpClient;
import com.openai.models.images.ImageGenerateParams;
import com.openai.models.images.ImageModel;
import java.io.IOException;

import com.openai.azure.credential.AzureApiKeyCredential; 


public class Sample {
    public static void main(String[] args) {

    String endpoint = "<%= openai_v1_endpoint %>";
    String apiKey = "{your-api-key}";
    String deploymentName = "<%= deploymentName %>";

        OpenAIClient client = OpenAIOkHttpClient.builder()
                .baseUrl(endpoint)
                .credential(AzureApiKeyCredential.create("{your-api-key}"))
                .build();

        ImageGenerateParams imageGenerateParams = ImageGenerateParams.builder()
                .responseFormat(ImageGenerateParams.ResponseFormat.URL)
                .prompt("A cute baby polar bear")
                .model(deploymentName)
                .size(ImageGenerateParams.Size._512X512)
                .n(1)
                .build();

        client.images().generate(imageGenerateParams).data().orElseThrow().stream()
                .flatMap(image -> image.url().stream())
                .forEach(System.out::println);
    }
}

File: generated-samples/go-v1/image-generation/key-auth/sample.go
package main

import (
	"context"
	"fmt"
	"log"
	"os"

	"github.com/openai/openai-go/v2"
	"github.com/openai/openai-go/v2/azure"
	"github.com/openai/openai-go/v2/option"
)

func main() {
	const endpoint = "<%= openai_v1_endpoint %>"
	const apiKey = "{your-api-key}"
	const deploymentName = "<%= deploymentName %>"

		client := openai.NewClient(
		option.WithBaseURL(endpoint),
		option.WithAPIKey(apiKey),
		)
	
	// Image URL
	image, err := client.Images.Generate(ctx, openai.ImageGenerateParams{
		Prompt:         "A cute baby polar bear",
		Model:          deploymentName,
		ResponseFormat: openai.ImageGenerateParamsResponseFormatURL,
		N:              openai.Int(1),
	})
	if err != nil {
		panic(err)
	}
	println("Image URL:")
	println(image.Data[0].URL)
	println()
}
